src/components/memorygame/MemoryGameFlippableCard/index.jsx

```diff
@@ -0,0 +1,80 @@
+import React from "react";
+// check https://github.com/AaronCCWong/react-card-flip for 'react-card-flip' details
+import ReactCardFlip from "react-card-flip";
+import { Card } from "antd";
+import questionMark from "../../../images/memoryGameFrontImage.png";
+import "./index.css";
+
+// targetState example
+// {
+//   cardKey: "alternateEnergy0",
+//   description: "Alternatif Enerji",
+//   img: {src: "/static/media/alternateEnergy.88cef00e.svg", imgKey: "alternateEnergy"},
+//   isFlipped: false,
+//   isMatched: false,
+//   link: "https://lorem.com",
+//   phrase: "Lorem ipsum"
+// }
+// FlippableCard is being used in a map function. So targetState refers to each iteration of map
+// Mapped array is cardStates. it holds states like example above(line 9-17)
+// selectedCards is an array of clicked card states(objects).
+// selectedCards array hold only "NUM_REQUIRED_MATCHES" amount of card states
```

Comment:
rather than writing this comment its better to write an invariant inside the function body that this is true, and you can delete this comment. let your code enforce this fact

but also i think this comment might be wrong, can't selectedCards be less than NUM_REQUIRED_MATCHES? you can have only one card selected. in that case, these comments or variable names need to be fixed

Reply:
Yes comment is wrong. It can hold less than "NUM_REQUIRED_MATCHES". What I meant was useEffect returns nothing if its length is less than "NUM_REQUIRED_MATCHES". I will fix it

Reply:
I have deleted it. invariant will be added to MemoryGameBoard component. PTAL